### PR TITLE
POC: Group by variable

### PR DESCRIPTION
**Pull request by dprokop**
Exploration of an alternative approach to https://github.com/grafana/scenes/pull/548. 

This brings aggreagations as a Scene Variable. Seems to be working fine, just had to do minor changes to the multi value variable interface in order allow clearing the value.

Todo:
- [ ] Demos update to show both static and dynamic version
- [x] Add variable tests

### CODE REVIEW

File: packages/scenes/src/variables/components/VariableValueSelect.tsx [hunk 1]
```diff
@@ -74,7 +74,10 @@ export function VariableValueSelectMulti({ model }: SceneComponentProps<MultiVal
       onBlur={() => {
         model.changeValueTo(uncommittedValue);
       }}
-      onChange={(newValue) => {
+      onChange={(newValue, action) => {
+        if (action.action === 'clear' && noValueOnClear) {
+          model.changeValueTo([]);
+        }
```
Comment (dprokop) [inline packages/scenes/src/variables/components/VariableValueSelect.tsx:80]: @torkelo  @ash this piece of code enables clearing the value of a variable. But, to preserve the current behavior of other multi value variables, it requries an explicit config of the variable `noValueOnClear`

File: packages/scenes/src/querying/SceneQueryRunner.ts [hunk 2]
```diff
@@ -442,7 +442,7 @@ export class SceneQueryRunner extends SceneObjectBase<QueryRunnerState> implemen
 
     if (this._adhocAggregationsSet) {
       // @ts-ignore (Temporary ignore until we update @grafana/data)
-      request.aggregations = this._adhocAggregationsSet.state.dimensions;
+      request.aggregations = this._adhocAggregationsSet.state.value;
```
Comment (torkelo) [inline packages/scenes/src/querying/SceneQueryRunner.ts:445]: this is a a confusing name to me, I thought it was request.groupBy or request.groupByDimensions ?  

aggregations to me imply aggregation functions, not dimensions. And if it did imply dimensions it is not clear if the dimensions should be excluded from aggregation (by) or included (without).

**Review by torkelo — COMMENTED** [review]
Looks good. Think skipping with the variable / set separation is the right thing, want to try to refactor and do the same to AdHocFilterSet. 

the only thing that does not feel 100% is the naming of the variable as AggregationsSetVariable. 

AggregationsSet feels like it would a set of aggregation functions (min, max, avg). 

Would not GroupByVariable make more sense? The request property we send to the data source was named groupBy in the design doc i I recall. I think naming this aggregations is very confusing.

File: packages/scenes/src/variables/components/VariableValueSelect.tsx (hunk 1 above)
Comment (torkelo) [reply packages/scenes/src/variables/components/VariableValueSelect.tsx:80]: looks good

**Review by ashharrison90 — APPROVED** [review]
think this is good, we can iterate 🤓

File: packages/scenes/src/querying/SceneQueryRunner.ts (hunk 2 above)
Comment (dprokop) [reply packages/scenes/src/querying/SceneQueryRunner.ts:445]: @torkelo  - just POC, we have not landed the required change in the core repo yet. I agree with you, `groupByDimensions` is probably the closes tho what it should be. I'll update as soon as we land this in core.

**Comment by dprokop** [comment]
> do we need to preserve the old behaviour, or could we just have this every time? 🤔

For now let's preserve, mainly fo backwards compatibility needs. We need to figure out empty state for variables as an independent thing.

File: packages/scenes/src/querying/SceneQueryRunner.ts (hunk 2 above)
Comment (torkelo) [reply packages/scenes/src/querying/SceneQueryRunner.ts:445]: @dprokop maybe we use groupByKeys instead to use the same word as in getTagKeys

**Review by torkelo — APPROVED** [review]

File: packages/scenes/src/variables/groupby/AggregationsSet.tsx [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import { SceneObjectBase } from '../../core/SceneObjectBase';
 import { AdHocVariableFilter, GrafanaTheme2, MetricFindValue, SelectableValue } from '@grafana/data';
-import { allActiveAggregationsSets } from './findActiveAggregationsSetByUid';
+import { allActiveAggregationsSets } from './findActiveGroupByVariablesByUid';
```
Comment (ashharrison90) [inline packages/scenes/src/variables/groupby/AggregationsSet.tsx:3]: tiny bit of renaming still to do here, but we can do that in the other branch once this is merged in 👍

**Review by ashharrison90 — APPROVED** [review]
lgtm, let's get this merged into the other branch so we can tidy everything up 😄

**Pull request merged**